File: backend/src/main/java/com/seuprojeto/progress/ProgressController.java
package com.seuprojeto.progress;

import com.seuprojeto.progress.dto.ProgressResponse;
import lombok.RequiredArgsConstructor;
import org.springframework.security.core.Authentication;
import org.springframework.web.bind.annotation.*;

@RestController
@RequestMapping("/api/progresso")
@RequiredArgsConstructor
public class ProgressController {

    private final ProgressService progressService;

    @PutMapping("/episodios/{episodioId}")
    public ProgressResponse marcarEpisodio(@PathVariable Long episodioId,
                                            @RequestParam ProgressStatus status,
                                            Authentication authentication) {
        return progressService.marcarEpisodio(episodioId, status, authentication.getName());
    }

    @PutMapping("/conteudos/{conteudoId}")
    public ProgressResponse marcarFilme(@PathVariable Long conteudoId,
                                         @RequestParam ProgressStatus status,
                                         Authentication authentication) {
        return progressService.marcarFilme(conteudoId, status, authentication.getName());
    }
}

File: backend/src/progress/progress.controller.ts
import { Controller, Param, ParseIntPipe, Put, Query } from '@nestjs/common';
import { ApiTags } from '@nestjs/swagger';
import { ProgressStatus } from '@prisma/client';
import { ProgressService } from './progress.service';
import { CurrentUser } from '../common/decorators/current-user.decorator';
import { User } from '@prisma/client';

@ApiTags('progress')
@Controller('api/progresso')
export class ProgressController {
  constructor(private readonly progressService: ProgressService) {}

  @Put('episodios/:episodioId')
  marcarEpisodio(
    @Param('episodioId', ParseIntPipe) episodioId: number,
    @Query('status') status: ProgressStatus,
    @CurrentUser() user: User,
  ) {
    return this.progressService.marcarEpisodio(episodioId, status, user.email);
  }

  @Put('conteudos/:conteudoId')
  marcarFilme(
    @Param('conteudoId', ParseIntPipe) conteudoId: number,
    @Query('status') status: ProgressStatus,
    @CurrentUser() user: User,
  ) {
    return this.progressService.marcarFilme(conteudoId, status, user.email);
  }
}
